dev/ci/ci-wrapper.sh

```diff
@@ -14,7 +15,31 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd "${DIR}/../.." || exit 1
 
 export TIMED=1
-bash "${DIR}/${CI_SCRIPT}" 2>&1 | tee "$CI_NAME.log"
+
+# NB: in CI TERM is unset in the environment
+# when TERM is unset, bash sets it to "dumb" as a bash variable (not exported?)
+if { [ -t 1 ] && ! [ "$TERM" = dumb ]; } || [ "$CI" ]
+then color_wanted=1
+else color_wanted=
+fi
+
+if [ "$color_wanted" ] && command -v script > /dev/null; then
+  # prevent piping from disabling auto colors / enable auto colors in CI
+    if [ "$CI" ]; then
+        export TERM=fake
+
+        # otherwise git log can get stuck waiting
+        # (but for some reason not always, eg https://gitlab.inria.fr/coq/coq/-/jobs/3638132
+        # vs failing jobs in that pipeline)
+        exec < /dev/null
```

Comment:
it still gets stuck in perennial, fiat crypto and fiat crypto legacy even with this

Why?
And why those jobs but not other WITH_SUBMODULES jobs?

Reply:
If it is `git log` that is stuck waiting, maybe you want `git --no-pager log`?